src/lib.rs

```diff
@@ -344,10 +350,9 @@ impl CspList {
             let directive = policy.directive_set.iter().find(|directive| directive.name == "base-uri");
             if let Some(directive) = directive {
                 if SourceList(&directive.value).does_url_match_source_list_in_origin_with_redirect_count(base, &self_origin, 0) == DoesNotMatch {
-                    let report_sample = directive.value.iter().any(|t| &t[..] == "'report-sample'");
```

Comment:
https://www.w3.org/TR/CSP/#allow-base-for-document has no step indicating we should check for `report-sample`